src/cluster_legacy.c

```diff
@@ -7041,3 +7049,21 @@ int clusterDecodeOpenSlotsAuxField(int rdbflags, sds s) {
 
     return C_OK;
 }
+
+void clusterCommandFlushslot(client *c) {
+    int slot;
+    int lazy = server.lazyfree_lazy_user_flush;
+    if ((slot = getSlotOrReply(c, c->argv[2])) == -1) return;
+    if (c->argc == 4) {
+        if (!strcasecmp(c->argv[3]->ptr, "async")) {
+            lazy = 1;
+        } else if (!strcasecmp(c->argv[3]->ptr, "sync")) {
+            lazy = 0;
+        } else {
+            addReplyErrorObject(c, shared.syntaxerr);
+            return;
+        }
+    }
```

Comment:
we should fail the `c->argc > 4` case I think

Reply:
In the call site of `clusterCommandFlushslot`, I have already performed a check on the number of parameters. Is it still necessary to validate the number of parameters again within the function?

```c
 else if (!strcasecmp(c->argv[1]->ptr, "flushslot") && (c->argc == 3 || c->argc == 4)) {
        /* CLUSTER FLUSHSLOT <slot> [ASYNC|SYNC] */
        clusterCommandFlushslot(c);
    }
```